### PR TITLE
Calendar idp helpers

### PR DESCRIPTION
**Pull request by mpkrass7**
## Summary
Add helpers for a calendar dataset an idp. It also modifies autopilot to add a calendar id if provided

### CODE REVIEW

File: src/datarobotx/idp/calendars.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        If a string, it must be in the format 'YYYY-MM-DD'.
+
+    Returns
+    -------
+    str
+        The ID of the created or retrieved calendar.
+    """
+    dr.Client(token=token, endpoint=endpoint)  # type: ignore
+
+    calendar_token = get_hash(name, country_code, start_date, end_date)
```
Comment (brau0300) [inline src/datarobotx/idp/calendars.py:61]: does our hasher work with dt.datetime objects?

Comment (brau0300) [reply src/datarobotx/idp/calendars.py:61]: looks like we're not testing that case

**Review by brau0300 — APPROVED** [review]
if hasher doesn't support dt.datetime will let's discuss further

File: src/datarobotx/idp/calendars.py (hunk 1 above)
Comment (mpkrass7) [reply src/datarobotx/idp/calendars.py:61]: No it does not work with datetime objects. I left the option of both in while we decide if we want to support

**Pull request merged**